backend/regalloc/regalloc_split.ml

```diff
@@ -0,0 +1,358 @@
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+open! Regalloc_utils
+open! Regalloc_split_utils
+module DLL = Flambda_backend_utils.Doubly_linked_list
+module State = Regalloc_split_state
+
+(* [propagate_substitution state cfg substs start subst] propagates the
+   substitution [subst], from block [start] until new ones are defined. The
+   effect of the propagation is the addition of [subst] to the relevant blocks
```

Comment:
No, it is not possible, I have augmented the comment
with the following:

```
   In practice, it means that we stop propagating when we encounter
   a node with either "definitions at beginning" or a "phi" because:
   - when there are new definitions, they follow a destruction point,
     which means that every live register will get a new name (thus
     replacing the binding in the current substitution);
   - when there is a phi, it means that every register in the substitution
     we are currently propagating is either no longer live, or needs a
     phi move (thus replacing the binding in the current substitution).
```